src/main/java/uk/ac/ebi/ampt2d/accession/AccessioningService.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.ampt2d.accession;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AccessioningService<T> {
+    private AccessionRepository<T> accessionRepository;
+
+    private AccessionGenerator<T> accessionGenerator;
+
+    public AccessioningService(AccessionRepository<T> accessionRepository,
+                               AccessionGenerator<T> accessionGenerator) {
+        this.accessionRepository = accessionRepository;
+        this.accessionGenerator = accessionGenerator;
+    }
+
+    public Map<T, String> createAccessions(List<T> objects) {
+        Map<T, String> accessions = new HashMap<>();
+
+        for (T object : objects) {
+            // look for an existing accession for this object
+            String accession = accessionRepository.get(object);
```

Comment:
This could become a big bottle neck in all the cases that require queries to database. Could this method be reformulated so that the retrieval of existing accessions is for a list and not for a single object?

I've seen this problem in the T2D when checking existing annotated elements and can slow by a large factor any process.